### PR TITLE
feat(Variance): Initial Props and Refactor Tests [WEB-1390]

### PR DESCRIPTION
**Pull request by codecaaron**
### [Variance Migration Doc](https://www.notion.so/codecademy/Variance-Migration-a675bbc31b584fec9dfff72c0364b194)

This PR is setting up for a cutover later but does not actually have any exposure. 

#### Gamut Changes
- Adds the base configuration for variance system props but does not export them yet.
- These should roughly be 1:1 with current gamut-system props.
- Checks in variance to kit, gamut, labs, and styles, but does not doing anything with them.

#### Variance Changes
- Refactors some of the tests of variance to make it more readable.
- Adds tests for `transformSize` and exports transforms from variance.
- Adds type overrides for `color` and `size` properties to ensure that we support the correct types for transforms and do not add overhead for aliased colors.

### PR Checklist

- [x] Related to designs: All of theme 😛 
- [x] Related to JIRA ticket:  WEB-1390
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.b13caf.0<br/>@codecademy/gamut-kit@0.2.48-alpha.b13caf.0<br/>@codecademy/gamut-labs@13.2.11-alpha.b13caf.0<br/>@codecademy/gamut-styles@9.1.5-alpha.b13caf.0<br/>@codecademy/gamut-tests@2.3.5-alpha.b13caf.0<br/>@codecademy/gamut@29.7.2-alpha.b13caf.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.b13caf.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.b13caf.0<br/>@codecademy/styleguide@33.6.3-alpha.b13caf.0<br/>@codecademy/variance@0.5.1-alpha.b13caf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b74c21e2c8f00cdd34cef--gamut-preview.netlify.app>https://606b74c21e2c8f00cdd34cef--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b74c21e2c8f00cdd34cef>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.a45b47.0<br/>@codecademy/gamut-kit@0.2.48-alpha.a45b47.0<br/>@codecademy/gamut-labs@13.2.11-alpha.a45b47.0<br/>@codecademy/gamut-styles@9.1.5-alpha.a45b47.0<br/>@codecademy/gamut-tests@2.3.5-alpha.a45b47.0<br/>@codecademy/gamut@29.7.2-alpha.a45b47.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.a45b47.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.a45b47.0<br/>@codecademy/styleguide@33.6.3-alpha.a45b47.0<br/>@codecademy/variance@0.5.1-alpha.a45b47.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b7873063de4008d48fd86--gamut-preview.netlify.app>https://606b7873063de4008d48fd86--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b7873063de4008d48fd86>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.5a6616.0<br/>@codecademy/gamut-kit@0.2.48-alpha.5a6616.0<br/>@codecademy/gamut-labs@13.2.11-alpha.5a6616.0<br/>@codecademy/gamut-styles@9.1.5-alpha.5a6616.0<br/>@codecademy/gamut-tests@2.3.5-alpha.5a6616.0<br/>@codecademy/gamut@29.7.2-alpha.5a6616.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.5a6616.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.5a6616.0<br/>@codecademy/styleguide@33.6.3-alpha.5a6616.0<br/>@codecademy/variance@0.5.1-alpha.5a6616.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b7aedfe0d7300c9c86adc--gamut-preview.netlify.app>https://606b7aedfe0d7300c9c86adc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b7aedfe0d7300c9c86adc>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.69741f.0<br/>@codecademy/gamut-kit@0.2.48-alpha.69741f.0<br/>@codecademy/gamut-labs@13.2.11-alpha.69741f.0<br/>@codecademy/gamut-styles@9.1.5-alpha.69741f.0<br/>@codecademy/gamut-tests@2.3.5-alpha.69741f.0<br/>@codecademy/gamut@29.7.2-alpha.69741f.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.69741f.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.69741f.0<br/>@codecademy/styleguide@33.6.3-alpha.69741f.0<br/>@codecademy/variance@0.5.1-alpha.69741f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606c6f4a6c228000d4b93584--gamut-preview.netlify.app>https://606c6f4a6c228000d4b93584--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606c6f4a6c228000d4b93584>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.8598e5.0<br/>@codecademy/gamut-kit@0.2.48-alpha.8598e5.0<br/>@codecademy/gamut-labs@13.2.11-alpha.8598e5.0<br/>@codecademy/gamut-styles@9.1.5-alpha.8598e5.0<br/>@codecademy/gamut-tests@2.3.5-alpha.8598e5.0<br/>@codecademy/gamut@29.7.2-alpha.8598e5.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.8598e5.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.8598e5.0<br/>@codecademy/styleguide@33.6.3-alpha.8598e5.0<br/>@codecademy/variance@0.5.1-alpha.8598e5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606c75721322f8008dd2911a--gamut-preview.netlify.app>https://606c75721322f8008dd2911a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606c75721322f8008dd2911a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.0f9b0a.0<br/>@codecademy/gamut-kit@0.2.48-alpha.0f9b0a.0<br/>@codecademy/gamut-labs@13.2.11-alpha.0f9b0a.0<br/>@codecademy/gamut-styles@9.1.5-alpha.0f9b0a.0<br/>@codecademy/gamut-tests@2.3.5-alpha.0f9b0a.0<br/>@codecademy/gamut@29.7.2-alpha.0f9b0a.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.0f9b0a.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.0f9b0a.0<br/>@codecademy/styleguide@33.6.3-alpha.0f9b0a.0<br/>@codecademy/variance@0.5.1-alpha.0f9b0a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606c7a59e929de0c587d5cf7--gamut-preview.netlify.app>https://606c7a59e929de0c587d5cf7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606c7a59e929de0c587d5cf7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.d5bae5.0<br/>@codecademy/gamut-kit@0.2.48-alpha.d5bae5.0<br/>@codecademy/gamut-labs@13.2.11-alpha.d5bae5.0<br/>@codecademy/gamut-styles@9.1.5-alpha.d5bae5.0<br/>@codecademy/gamut-tests@2.3.5-alpha.d5bae5.0<br/>@codecademy/gamut@29.7.2-alpha.d5bae5.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.d5bae5.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.d5bae5.0<br/>@codecademy/styleguide@33.6.3-alpha.d5bae5.0<br/>@codecademy/variance@0.5.1-alpha.d5bae5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606c7c9b854a1f10e79efc02--gamut-preview.netlify.app>https://606c7c9b854a1f10e79efc02--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606c7c9b854a1f10e79efc02>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.7c8f13.0<br/>@codecademy/gamut-kit@0.2.48-alpha.7c8f13.0<br/>@codecademy/gamut-labs@13.2.11-alpha.7c8f13.0<br/>@codecademy/gamut-styles@9.1.5-alpha.7c8f13.0<br/>@codecademy/gamut-tests@2.3.5-alpha.7c8f13.0<br/>@codecademy/gamut@29.7.2-alpha.7c8f13.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.7c8f13.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.7c8f13.0<br/>@codecademy/styleguide@33.6.3-alpha.7c8f13.0<br/>@codecademy/variance@0.5.1-alpha.7c8f13.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606c7fbf40719b12b942a90d--gamut-preview.netlify.app>https://606c7fbf40719b12b942a90d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606c7fbf40719b12b942a90d>Deploy Logs</a></p>

File: packages/variance/src/transforms/__tests__/transformSize-test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ];
+
+  absoluteValues.forEach(([value, result]) => {
+    it(`transforms absolute value of ${value} to the pixel value of ${result}`, () => {
+      expect(percentageOrAbsolute(value)).toEqual(result);
+    });
+  });
+});
+
+describe(transformSize, () => {
```
Comment (JoshuaKGoldberg) [inline packages/variance/src/transforms/__tests__/transformSize-test.ts:36]: You might enjoy `it.each` for these...

**Review by JoshuaKGoldberg — APPROVED** [review]
Makes sense.

Reiterating but not blocking on a concern here: I'm not big on shorthands like `p` and `mx` and suspect this'll be confusing for the team. But for the sake of shipping fast & iteratively and gathering user data, I think it's very good to at least try them out!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.8-alpha.58ed08.0<br/>@codecademy/gamut-kit@0.2.48-alpha.58ed08.0<br/>@codecademy/gamut-labs@13.2.11-alpha.58ed08.0<br/>@codecademy/gamut-styles@9.1.5-alpha.58ed08.0<br/>@codecademy/gamut-tests@2.3.5-alpha.58ed08.0<br/>@codecademy/gamut@29.7.2-alpha.58ed08.0<br/>@codecademy/konamimojisplosion@0.2.5-alpha.58ed08.0<br/>@codecademy/markdown-overrides@0.5.18-alpha.58ed08.0<br/>@codecademy/styleguide@33.7.1-alpha.58ed08.0<br/>@codecademy/variance@0.5.1-alpha.58ed08.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606cb179e27cb41fa8ccb01d--gamut-preview.netlify.app>https://606cb179e27cb41fa8ccb01d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606cb179e27cb41fa8ccb01d>Deploy Logs</a></p>

**Pull request merged**